nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentNotification.java

```diff
@@ -97,56 +97,84 @@ protected CommentManager getCommentManager() {
 
     @Test
     public void shouldNotifyEventWhenCreateComment() {
-        // We subscribe to the creation document to check that we will not be notified about the comment creation as
-        // document (see CommentCreationVeto), only the comment added, and the 'File' document creation
-        captureAndVerifyCommentEventNotification(() -> {
-            Comment createdComment = createCommentAndAddSubscription("CommentAdded", "Creation");
-            return session.getDocument(new IdRef(createdComment.getId()));
-        }, COMMENT_ADDED, DOCUMENT_CREATED);
+        try (CapturingEventListener listener = new CapturingEventListener(COMMENT_ADDED)) {
+            Comment comment = createComment(commentedDocumentModel);
+            DocumentModel commentDocumentModel = session.getDocument(new IdRef(comment.getId()));
+            DocumentModel commentParentDocumentModel = session.getDocument(
+                    new IdRef((String) commentDocumentModel.getPropertyValue(COMMENT_PARENT_ID)));
+            transactionalFeature.nextTransaction();
```

Comment:
Generally, put the `transactionalFeature.nextTransaction();` just after the code triggering async stuff, and then fetch new documents with the new session.

Same for all tests in all files in this PR.